### PR TITLE
Refactor: $BUILD_DIR

### PR DESCRIPTION
**Pull request by CaseyFaist**
In most places (aside from possibly simlinks) /app/ is the same as the $BUILD_DIR var used throughout the buildpack. 

@duanehutchins refactored to reflect this, making the buildpack cleaner and more consistent  🎉

This pr supercedes #761 so I can run tests with creds

### CODE REVIEW

**Comment by CaseyFaist** [comment]
This will need to be rebased against #866 (changelog fixes), waiting on that.

**Comment by movermeyer** [comment]
@schneems @CaseyFaist 

Our downstream build has starting failing. We believe it is a result of this PR.

```
ln: failed to create symbolic link '/tmp/build/.heroku/python': No such file or directory
```

It seems that this doesn’t create `.heroku/python` before doing this:
https://github.com/heroku/heroku-buildpack-python/pull/868/files#diff-c7b43338c09c3c9afd420fd8b6bb5198R225

Please let me know if there is more I can do to help debug the problem.

**Comment by luyizhao** [comment]
L108 of `bin/compile` at https://github.com/heroku/heroku-buildpack-python/pull/868/files#diff-c7b43338c09c3c9afd420fd8b6bb5198R108 seems to be missing a `$`. This might be causing the issues.

**Comment by cw-intellineers** [comment]
Have the same issue as the previous people.

**Comment by hugorodgerbrown** [comment]
Same here - review apps now failing to build.

**Comment by dzuelke** [comment]
This must be reverted.

File: bin/compile [hunk 1]
```diff
@@ -104,7 +105,7 @@ source "$BIN_DIR/warnings"
 # to `/app`.
 # Symlinks are required, since Python is not a portable installation.
 # More on this topic later.
-mkdir -p /app/.heroku
+mkdir -p "BUILD_DIR/.heroku"
```
Comment (dzuelke) [inline bin/compile]: this is faulty: missing a `$`

**Comment by ulkoart** [comment]
> ln: failed to create symbolic link '/tmp/build/.heroku/python': No such file or directory

error still reproducing.

**Pull request merged**